pkg/sql/opt/memo/multiplicity_builder.go

```diff
@@ -118,8 +118,10 @@ func deriveUnfilteredCols(in RelExpr) opt.ColSet {
 		md := t.Memo().Metadata()
 		baseTable := md.Table(t.Table)
 		if t.IsUnfiltered(md) {
-			for i, cnt := 0, baseTable.ColumnCount(); i < cnt; i++ {
-				unfilteredCols.Add(t.Table.ColumnID(i))
+			for i, cnt := 0, baseTable.AllColumnCount(); i < cnt; i++ {
+				if !cat.IsMutationColumn(baseTable, i) {
```

Comment:
I don't see any harm in adding mutation cols here. I think Drew just used `ColumnCount` b/c it seemed like the right thing. But really we want all columns.